src/directions/events.ts

```diff
@@ -8,44 +8,45 @@ export class MapLibreGlDirectionsEvented {
 
   protected readonly map: Map;
 
-  private listeners: Partial<Record<keyof MapLibreGlDirectionsEventType, ((ev: never) => void)[]>> = {};
-  private oneTimeListeners: Partial<Record<keyof MapLibreGlDirectionsEventType, ((ev: never) => void)[]>> = {};
+  private listeners: ListenersStore = {};
+  private oneTimeListeners: ListenersStore = {};
 
   protected fire<T extends keyof MapLibreGlDirectionsEventType>(event: MapLibreGlDirectionsEventType[T]) {
     event.target = this.map;
 
-    this.listeners[event.type]?.forEach((listener) => listener(event));
-    this.oneTimeListeners[event.type]?.forEach((listener) => {
+    const type: T = event.type as T;
+
+    this.listeners[type]?.forEach((listener) => listener(event));
+    this.oneTimeListeners[type]?.forEach((listener) => {
       listener(event);
 
-      const index = this.oneTimeListeners[event.type]?.indexOf(listener);
-      if (index && ~index) this.oneTimeListeners[event.type]?.splice(index, 1);
+      const index = this.oneTimeListeners[type]?.indexOf(listener);
+      if (index !== undefined && ~index) this.oneTimeListeners[type]?.splice(index, 1);
```

Comment:
Why explicitly checking for `undefined`? Shouldn't a simple falsy-assertion be enough (`!index`)?

Reply:
`!index` will be falsy for truthy values which is the opposite and `!!index` which might do as non-nil check in some other cases also won't work as it doesn't handle case with 0.

Reply:
True, agreed.